### ISSUE TITLE
cov_pca & tetra_pca: TypeError: an integer is require

### ISSUE THREAD

**Issue by Thomieh73** [Ecogenomics/CheckM#153]
Hi Donovan,
I opened an issue but it matches closed issue 109. But I did not see a solution there.

I have an issue when running either cov_pca or tetra_pca.

I have a checkm installation running in a conda environment on a computing cluster. Checkm version 1.0.11.

I got the same error when running the commands on my metagenome bins or on a fasta file with a bacterial genome. For the later I used the genome of Thermosipho africanus TCF52B.

These were the commands for tetra_pca:
```
checkm tetra -t 32 test_bins/Tafricanus_TCF52B_NC_011653.fna test_bins_tetra_freq.tsv

checkm tetra_pca -x fna test_bins test_bins_plots test_bins_tetra_freq.tsv
```
That produced this output:
```
Unexpected error: <type 'exceptions.TypeError'>
Traceback (most recent call last):
  File "/usit/abel/u1/thhaverk/tools/anaconda3/envs/checkm_env/bin/checkm", line 709, in <module>
    checkmParser.parseOptions(args)
  File "/usit/abel/u1/thhaverk/tools/anaconda3/envs/checkm_env/lib/python2.7/site-packages/checkm/main.py", line 1286, in parseOptions
    self.tetraPcaPlot(options)
  File "/usit/abel/u1/thhaverk/tools/anaconda3/envs/checkm_env/lib/python2.7/site-packages/checkm/main.py", line 565, in tetraPcaPlot
    seqIds, pc, variance = pca.pcaFile(options.tetra_profile, fraction=1.0, bCenter=True, bScale=False)
  File "/usit/abel/u1/thhaverk/tools/anaconda3/envs/checkm_env/lib/python2.7/site-packages/checkm/PCA.py", line 53, in pcaFile
    self.pcaMatrix(data, fraction, bCenter, bScale)
  File "/usit/abel/u1/thhaverk/tools/anaconda3/envs/checkm_env/lib/python2.7/site-packages/checkm/PCA.py", line 62, in pcaMatrix
    Center(A, bScale)
  File "/usit/abel/u1/thhaverk/tools/anaconda3/envs/checkm_env/lib/python2.7/site-packages/checkm/PCA.py", line 112, in __init__
    self.mean = A.mean(axis=axis)
  File "/usit/abel/u1/thhaverk/tools/anaconda3/envs/checkm_env/lib/python2.7/site-packages/numpy/core/_methods.py", line 70, in _mean
    ret = umr_sum(arr, axis, dtype, out, keepdims)
TypeError: an integer is required
```

Any idea on what is happening here and how to solve it?

btw. I did notice the same issue when running `refinem outliers` and asking for plots.

**Comment by dparks1134** [comment]
Hello. Best guess is that this is due to recent changes with matplotlib. I use version 1.3.1 on my system. I will look to test CheckM under the latest version of matplotlib, but it will likely take me a bit of time to get to this. Sorry for the inconvenience.

**Comment by Thomieh73** [comment]
Hi Donovan,
Thanks for pointing out which matplotlib you use. I have now created a new conda environment for checkm, where I first install matplotlib (1.3.1) and then I install checkm. After that I can run the cov_pca & tetra_pca without any problems. 

Below is the conda installation instructions I have used:

Installing a python 2.7 environment with conda and all the scientific anaconda packages
```
conda create -n checkm_env python=2.7
```
Activate the conda environment
```
conda activate checkm_env
```
Testing the python version and the pip version installed
```
which python
which pip
```
Installing an old version of matplotlib (v 1.3.1)
```
conda install matplotlib=1.3.1
```
Now use conda to install the latest checkm version using a virtual environment
```
conda install -c bioconda checkm-genome 
```
Trying if we can call the checkm program
```
checkm
```
If that creates no error messages than  checkm is installed

Deactivate the conda environment
```
conda deactivate
```
